### PR TITLE
[4.0] Load component language files in API

### PR DESCRIPTION
**Pull request by SharkyKZ**
Partial Pull Request for https://github.com/joomla/joomla-cms/issues/26846.

### Summary of Changes

Corrects component name detection in `Joomla\CMS\Component\ComponentHelper::getComponentName()` when using API components without Dispatcher.

Corrects path to language files.

### Testing Instructions

Cause an error when saving a contact with API. E.g. try to save a contact with duplicate alias.

### Expected result

>Save failed with the following error: Another Contact from this category has the same alias (remember it may be a trashed item).

### Actual result

>Save failed with the following error: COM_CONTACT_ERROR_UNIQUE_ALIAS

### Documentation Changes Required

No.

### CODE REVIEW

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 74df4a7d4b2cab1c1910878894106c6bab9b9d3b<br /><br />Tested successful. Received correct error message when creating a contact with the same alias. Tested other contact changes such as deleting and editing with success.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27593">issues.joomla.org/tracker/joomla-cms/27593</a>.</sub>

File: libraries/src/Dispatcher/ApiDispatcher.php [hunk 1]
```diff
@@ -65,7 +65,7 @@ final class ApiDispatcher extends ComponentDispatcher
 	protected function loadLanguage()
 	{
 		// Load common and local language files.
-		$this->app->getLanguage()->load($this->option, JPATH_BASE, null, false, true) ||
+		$this->app->getLanguage()->load($this->option, JPATH_ADMINISTRATOR, null, false, true) ||
```
Comment (wilsonge) [inline libraries/src/Dispatcher/ApiDispatcher.php]: Let's add this as a second line please. I'd like to prioritise JPATH_BASE if it exists (i.e. the api directory)

Comment (SharkyKZ) [reply libraries/src/Dispatcher/ApiDispatcher.php]: Added.

**Comment by wilsonge** [comment]
Nice work! Thankyou!

**Pull request merged**